torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -333,11 +333,34 @@ class ProcessGroupNCCL : public ProcessGroup {
       return !value_.isNone();
     }
 
+    std::shared_ptr<at::cuda::CUDAStream> getfutureNCCLCallbackStream() {
+      return futureNCCLCallbackStream_;
+    }
+
+    bool hasRecordStreamCallback() const {
+        return record_stream_cb_ != nullptr;
+    }
+
+    void setRecordStreamCallback(
+        std::function<void(const at::IValue&, at::cuda::CUDAStream)>
+            record_stream_cb) {
+      record_stream_cb_ = std::move(record_stream_cb);
+    }
+
+    void recordStream() {
+      TORCH_CHECK(record_stream_cb_);
+      // Do not free the underlying data storage of value_ before its usage on
+      // futureNCCLCallbackStream_ finish.
+      record_stream_cb_(value_, *futureNCCLCallbackStream_);
+    }
```

Comment:
You probably don't need a separate method for this if you're doing it in `addCallback()` instead.